test/e2e/multicast.go

```diff
@@ -202,19 +202,36 @@ func testMulticastIGMPQuery(f *framework.Framework, clientNodeInfo, serverNodeIn
 		framework.Failf("failed to retrieve multicast IGMP query: " + err.Error())
 	}
 
-	// The multicast listener pod join multicast group (-B 224.1.1.1), UDP (-u), during (-t 30) seconds, report every (-i 5) seconds
-	ginkgo.By("multicast listener pod join multicast group")
-	e2ekubectl.RunKubectl(f.Namespace.Name, "exec", multicastListenerPod, "--", "/bin/bash", "-c", fmt.Sprintf("iperf -s -B %s -u -t 30 -i 5", mcastGroup))
+	ginkgo.By("multicast listener pod joins multicast group and verifies IGMP query")
+	maxDuration := 2 * time.Minute
+	start := time.Now()
 
-	ginkgo.By(fmt.Sprintf("verifying that the IGMP query has been received"))
-	kubectlOut, err := e2ekubectl.RunKubectl(f.Namespace.Name, "exec", multicastListenerPod, "--", "/bin/bash", "-c", fmt.Sprintf("cat %s | grep igmp", tcpdumpFileName))
-	if err != nil {
-		framework.Failf("failed to retrieve multicast IGMP query: " + err.Error())
-	}
-	framework.Logf("output:")
-	framework.Logf(kubectlOut)
-	if kubectlOut == "" {
-		framework.Failf("failed to retrieve multicast IGMP query: igmp messages on the tcpdump logfile not found")
+	for {
+		elapsed := time.Since(start)
+		if elapsed > maxDuration {
+			framework.Failf("Failed to retrieve multicast IGMP query within %v", maxDuration)
+			break
+		}
+
+		ginkgo.By("Running iperf to join the multicast group")
```

Comment:
@flavio-fernandes This part of the test wasn't actually needed at all.  The test is about checking whether ovn-kubernetes (through ovn-controller) generates IGMP Queries periodically.  That should happen regardless of pods joining specific groups or not.

Which also makes me wonder, where was this flake seen, do you have a link to the logs?  I didn't see it in upstream CI but I also didn't follow all runs closely.

Reply:
hi @dceara ! Not sure I see the failure upstream anymore. Will check. I mostly see them downstream now.
Valid point about the iperf step. I think it was added to add a 30 seconds delay, which (in most cases?) is enough to get the periodic igmp queries. On the same token, are you sure that having an app binding to the multicast address not a trigger for igmp messages to happen sooner? I suspected it, but did not dig deep enough to confirm it.

Reply:
@qinqon : assuming you were the original implementer of this test, would you know? ^^

Reply:
Rewrote this logic to just wait for the igmp packet

Reply:
@flavio-fernandes ovn-controller IGMP queries are periodic and are flooded to all logical ports of the switch, regardless of who joined multicast groups or not.